lib/drush-node.js

```diff
@@ -104,5 +106,84 @@ Drush.exec = function (command, options) {
   return def.promise;
 };
 
+/**
+ * Spawn a drush command.
+ *
+ * @param {array} args
+ *   An array of arguments to append to the command, e.g. ['cc', 'drush'].
+ * @param {object} options
+ *   A hash of options to add to the command, can contain:
+ *     - alias: the drush alias, e.g. "@self" to execute the command with.
+ *     - simulate: boolean, simulates all relevant actions.
+ *     - uri: the URI of the drupal site to use.
+ *     - echo: text to echo to the drush command.
+ *     - cat: a file to cat to the drush command.
+ *
+ * @return Promise
+ */
+Drush.spawn = function (args, options) {
+  args = args || [];
+  options = options || {};
+  var def = new Deferred();
+  var pipeToDrush = function (command, pipeArgs) {
+    var cmd = spawn(command, pipeArgs);
+    cmd.stdout.on('data', function (data) {
+      cmd.stdin.write(data);
+    });
+    cmd.stderr.on('data', function (data) {
+      console.log('' + data);
+    });
+    cmd.on('close', function (code) {
+      if (code !== 0) {
+        return def.reject(command + ' process exited with code ' + code);
```

Comment:
Scratch that. It should not resolve the promise. Since the purpose of this function is to pipe data to the drush command, the promise should not be resolved until after the drush command completes. We want to keep the rejection though since, if there was a problem with the piping of data, we don't want to continue on to the drush command. The rejection should probably be handled a little cleaner though. I'll work on that.